portal-frontend/src/app/features/create-application/create-application.component.ts

```diff
@@ -0,0 +1,80 @@
+import { Component, OnInit } from '@angular/core';
+import { FormControl, FormGroup, Validators } from '@angular/forms';
+import { Router } from '@angular/router';
+import { Observable, startWith, map } from 'rxjs';
+import { ApplicationService } from '../../services/application/application.service';
+import { LocalGovernmentDto } from '../../services/code/code.dto';
+import { CodeService } from '../../services/code/code.service';
+import { FileHandle } from '../../shared/file-drag-drop/drag-drop.directive';
+
+@Component({
+  selector: 'app-create-application',
+  templateUrl: './create-application.component.html',
+  styleUrls: ['./create-application.component.scss'],
+})
+export class CreateApplicationComponent implements OnInit {
+  applicantName = new FormControl<string | any>('', [Validators.required]);
+  localGovernment = new FormControl<string | any>('', [Validators.required]);
+
+  localGovernments: LocalGovernmentDto[] = [];
+  filteredLocalGovernments!: Observable<LocalGovernmentDto[]>;
+
+  firstForm = new FormGroup({
+    applicantName: this.applicantName,
+    localGovernment: this.localGovernment,
+  });
+
+  files: FileHandle[] = [];
+
+  constructor(
+    private applicationService: ApplicationService,
+    private codeService: CodeService,
+    private router: Router
+  ) {}
+
+  filesDropped(files: FileHandle[]): void {
+    this.files = files;
+  }
+
+  ngOnInit(): void {
+    this.loadCodes();
+    this.filteredLocalGovernments = this.localGovernment.valueChanges.pipe(
+      startWith(''),
+      map((value) => this.filter(value || ''))
+    );
+  }
+
+  private filter(value: string): LocalGovernmentDto[] {
+    if (this.localGovernments) {
+      const filterValue = value.toLowerCase();
+      return this.localGovernments.filter((localGovernment) =>
+        localGovernment.name.toLowerCase().includes(filterValue)
+      );
+    }
+    return [];
+  }
+
+  private async loadCodes() {
+    const codes = await this.codeService.loadCodes();
+    this.localGovernments = codes.localGovernments;
+  }
+
+  async onSaveAndExit() {
+    const localGovernmentName = this.localGovernment.getRawValue();
```

Comment:
What is the flow in case the save fails? Is it a later thing?

Reply:
I'm not sure, nothing in the requirements